### PR TITLE
Remove project configuration from catalog

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR removes `project` entries from vcm.catalog.catalog.

- [ ] Tests added

Coverage reports (updated automatically):
- test_unit: [82%](https://output.circle-artifacts.com/output/job/98c8a307-5d50-427a-962d-338a0e3d479b/artifacts/0/tmp/coverage/htmlcov-test_unit/index.html)

### CODE REVIEW

File: external/vcm/vcm/catalog.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ sources:
     driver: zarr
     args:
       storage_options:
-        project: 'vcm-ml'
+
```
Comment (frodre) [inline external/vcm/vcm/catalog.yaml:10]: Did you mean to leave an empty line in all of these?

File: external/vcm/vcm/catalog.yaml [hunk 2]
```diff
@@ -716,7 +716,7 @@ sources:
         - northward_wind
     args:
       storage_options:
-        project: 'vcm-ml'
+
```
Comment (frodre) [inline external/vcm/vcm/catalog.yaml:719]: Leftover `project: 'vcm-ml'` on line 698

**Review by frodre — APPROVED** [review]
I tested on my VM that it worked to load some of these with the project variable removed.

File: external/vcm/vcm/catalog.yaml (hunk 1 above)
Comment (mcgibbon) [reply external/vcm/vcm/catalog.yaml]: Fixed. This was a side-effect of find-and-replace.

File: external/vcm/vcm/catalog.yaml (hunk 2 above)
Comment (mcgibbon) [reply external/vcm/vcm/catalog.yaml]: Fixed.

**Pull request merged**